pyvista/core/dataset.py

```diff
@@ -1420,6 +1420,7 @@ def copy_meta_from(self, ido: 'DataSet'):
         self._active_vectors_info = ido.active_vectors_info
         self.clear_textures()
         self._textures = {name: tex.copy() for name, tex in ido.textures.items()}
+        self._association_complex_names = ido._association_complex_names.copy()
```

Comment:
Either https://github.com/pyvista/pyvista/pull/2779 or this should be merged first, and this extended for handling deep or shallow copies on the other PR. I recommend merging 2779 first (in time) and then handling this.

Reply:
Done in c2322ac1662ed7c18193f5dffef037203b95459a.

Reply:
Crazy that we didn't catch this earlier, but I guess people weren't using bit arrays that often. With complex it might be more of an issue.